src/snmp.c

```diff
@@ -1754,11 +1778,18 @@ static int csnmp_read_table(host_definition_t *host, data_definition_t *data) {
       continue;
     }
 
-    for (vb = res->variables, i = 0; (vb != NULL);
-         vb = vb->next_variable, i++) {
+    for (vb = res->variables, j = 0; (vb != NULL);
```

Comment:
since you're only using j in this loop, please declare it here. Not all existing code uses C99, but new code should.

Reply:
I fear it is not possible since the loop has to initialize two variables it is not possible to use local declaration (well not without using a struct to hold j and vb)

Reply:
Ah right, in that case can you declare it just above the loop to reduce the scope?